### PR TITLE
feat: add path of incompatible versions

### PR DESCRIPTION
**Pull request by OliverGeneser**
:heart: Thank you for your contribution!

### CODE REVIEW

**Comment by utnapischtim** [comment]
question: may i ask what the motivation for this PR is?

File: pywebpack/helpers.py [hunk 1]
```diff
@@ -112,17 +112,23 @@ def max_version(v1, v2):
     return v1 if v1_maj > v2_maj else v2
 
 
-def merge_deps(computed_deps, incoming_deps):
+def merge_deps(computed_deps, incoming_deps, incoming_label=None, origins=None):
```
Comment (ntarocco) [inline pywebpack/helpers.py]: I find a bit obscure the way `origins` is used, by changing its value at each loop iterations.
Would it be better to pass it, and return it as extra return arg instead? Same effect, a bit more explicit.

**Review by ntarocco — APPROVED** [review]

**Comment by ntarocco** [comment]
> question: may i ask what the motivation for this PR is?

As someone who worked with this module quite heavily in the past, I am pretty sure Oliver took the **very good initiative** to improve the log output to make debugging easier.

**Comment by OliverGeneser** [comment]
> question: may i ask what the motivation for this PR is?

@utnapischtim the motivation was that while I've been doing the React 18 upgrade, it's been really hard to find which module was causing the major version mismatch... Now you can see the path! 😃

**Comment by utnapischtim** [comment]
@OliverGeneser yeah i thought that this will be the reason. i really like your improvement!

File: pywebpack/helpers.py (hunk 1 above)
Comment (OliverGeneser) [reply pywebpack/helpers.py]: I agree, but I chose to follow the same pattern we use for computed_deps because that is also mutated 🤔 Should we also change it for computed_deps then?

File: pywebpack/helpers.py [hunk 2]
```diff
@@ -127,17 +149,30 @@ def merge_deps(computed_deps, incoming_deps):
...
+                    computed_origin = ""
+                    if (dep_type, incoming_pkg) in origins:
+                        computed_origin = f" (from {origins[(dep_type, incoming_pkg)]})"
+
                     v_maj, _, _, _ = _parse_version(incoming_version)
                     tv_maj, _, _, _ = _parse_version(computed_version)
                     if v_maj != tv_maj:
                         raise MergeConflictError(
-                            f"Incompatible major versions for package {incoming_pkg}: current version is {computed_version}, incoming version is {incoming_version}"
+                            f"Incompatible major versions for package "
```
Comment (utnapischtim) [inline pywebpack/helpers.py]: nit: following the ruff rule [EM102](https://docs.astral.sh/ruff/rules/f-string-in-exception/) the f-string should be outside of the Exception.

File: pywebpack/helpers.py [hunk 3]
```diff
@@ -106,17 +112,33 @@ def max_version(v1, v2):
...
+        package, keyed by `(dep_type, pkg_name)`.
+    :param return_origins: returns `(omputed_deps, origins)` if true.
+    :return: the computed_deps dict with merged `incoming_deps` into it,
+        or a tuple with the updated origins when `return_origins` is set.
     :raise: raises the MergeConflictError exception when deps
         cannot be merged.
     """
+    if origins is None:
+        origins = {}
+    incoming_src = f" (from {incoming_label})" if incoming_label else ""
```
Comment (utnapischtim) [inline pywebpack/helpers.py]: opinion: personally i would move that to where it is used, so to `raise MergeConflictError`

**Comment by utnapischtim** [comment]
for me it looks good, but i would like to test this first, before i approve it.

**Comment by OliverGeneser** [comment]
> for me it looks good, but i would like to test this first, before i approve it.

Please do and let me know if you hit any weird edge cases :blush:

**Comment by utnapischtim** [comment]
> > for me it looks good, but i would like to test this first, before i approve it.
> 
> Please do and let me know if you hit any weird edge cases 😊

i tried it and it works perfectly. 

you could improve the `MergeConflictError` presentation a little bit. With long paths to the directory it reads not nicely, but that is not a blocker from my side

**Review by utnapischtim — APPROVED** [review]
